src/components/results/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { IAppStateExtended } from '@/app';
+import { pokemonApi } from '@/services';
+import React from 'react';
+import { PokemonCard } from '@/components';
+import { IPokemon } from '@/types';
+
+type ResultsProps = Pick<
+  IAppStateExtended,
+  'changeStatus' | 'searchQuery' | 'status'
+>;
+
+type ResultsState = { result: IPokemon | (IPokemon | null)[] | null };
+
+export class Results extends React.PureComponent<ResultsProps, ResultsState> {
+  controllerRef = React.createRef<AbortController>();
+
+  state: ResultsState = {
+    result: null,
+  };
+
+  loadResults = async () => {
+    this.props.changeStatus('loading');
```

Comment:
better to create variable for app states to avoid using the same string in multi places. ANd in future is convenient to change in one place.